include/drivers/pcie/pcie.h

```diff
@@ -124,17 +124,22 @@ extern bool pcie_probe_mbar(pcie_bdf_t bdf,
  */
 extern void pcie_set_cmd(pcie_bdf_t bdf, uint32_t bits, bool on);
 
+#ifndef CONFIG_PCIE_CONTROLLER
 /**
  * @brief Allocate an IRQ for an endpoint.
  *
  * This function first checks the IRQ register and if it contains a valid
  * value this is returned. If the register does not contain a valid value
  * allocation of a new one is attempted.
+ * Such function is only exposed if CONFIG_PCIE_CONTROLLER is unset.
+ * It is thus available only in x86, for now, where dynamic IRQ allocation
+ * for PCIe device makes sense.
```

Comment:
I'd say "where architecture tied dynamic IRQ allocation for PCIe device makes sense" because we also need dynamic IRQ allocation for ARM, but it's not tied to arch but to the MSI translater either tied to the IRQ controller or inside the PCIe controller itself

Reply:
Ok